roles/tests/test_user_models.py

```diff
@@ -51,3 +51,11 @@ def __init__(self, foo_var):
 
         user_group2 = FooObject('bar')
         self.assertFalse(user_group1.__eq__(user_group2))
+
+    def test_user_group_hash(self):
+        """Test for UserGroup __hash__"""
+        user_group = UserGroup('foo1', 'bar')
+        self.assertEqual(
+            user_group.__hash__(),
+            hash((user_group.username, user_group.group_type))
```

Comment:
Yeah, I see your point. The extra tests aren't necessary. I do like seeing apples compared to apples, which is why I would like to see `__hash__` on two separate `UserGroup` instances compared instead of `hash(('string' ,' string'))`, because otherwise it kind of _is_ testing `hash()`.

I think what I was shooting for with the "change string" tests is that if the `__hash__` method is ever changed, that test will break and that would be an indicator that we needed to search the codebase looking for our own uses of `hash()` to make sure we updated them, but that should be obvious during the development and code review.

So, how about just my first comment (instantiating another `UserGroup`, and ignoring the other parts?